### PR TITLE
[Docs] Refactoring Docker Container Docs

### PR DESCRIPTION
**Pull request by cblmemo**
Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: docs/source/examples/docker-containers.rst [hunk 1]
```diff
@@ -3,16 +3,36 @@
 Using Docker Containers
 =======================
 
-SkyPilot can run a Docker container either as the runtime environment for your task, or as the task itself.
+SkyPilot can run a Docker container either as a containerized application, or the runtime environment for your task.
```
Comment (concretevitamin) [inline docs/source/examples/docker-containers.rst:6]:
```suggestion
SkyPilot can run a Docker container either as a containerized application, or the runtime environment of a cluster.
```

File: docs/source/examples/docker-containers.rst [hunk 2]
```diff
@@ -3,16 +3,36 @@
...
+
+Running Containerized Application
+---------------------------------
```
Comment (concretevitamin) [inline docs/source/examples/docker-containers.rst:9]:
```suggestion
Running Containerized Applications
----------------------------------
```

File: docs/source/examples/docker-containers.rst [hunk 3]
```diff
@@ -3,16 +3,36 @@
...
+
+SkyPilot is capable of running containerized application directly. The Docker runtime comes pre-configured and ready for use on the default VM image provided by SkyPilot, making it convenient for executing containerized applications.
```
Comment (concretevitamin) [inline docs/source/examples/docker-containers.rst:11]:
```suggestion
SkyPilot is capable of running containerized applications directly. The Docker runtime comes pre-configured and ready for use on the default VM images provided by SkyPilot.
```

File: docs/source/examples/docker-containers.rst [hunk 4]
```diff
@@ -3,16 +3,36 @@
 Using Docker Containers
 =======================
 
-SkyPilot can run a Docker container either as the runtime environment for your task, or as the task itself.
+SkyPilot can run a Docker container either as a containerized application, or the runtime environment for your task.
+
+Running Containerized Application
```
Comment (concretevitamin) [inline docs/source/examples/docker-containers.rst:8]: Should we mention anything about private registries for this first case?

File: docs/source/examples/docker-containers.rst [hunk 5]
```diff
@@ -3,16 +3,36 @@
...
 Using Docker Containers as Runtime Environment
 ----------------------------------------------
 
-When a container is used as the runtime environment, the SkyPilot task is executed inside the container.
+When a container is used as the runtime environment, all the SkyPilot task is executed inside the container.
+
+This means all :code:`setup` and :code:`run` commands in the YAML file will be executed in the container, and any files created by the task will be stored inside the container. The Docker container functions as an image ID within the virtual machines provided by cloud service providers.
 
-This means all :code:`setup` and :code:`run` commands in the YAML file will be executed in the container, and any files created by the task will be stored inside the container.
 Any GPUs assigned to the task will be automatically mapped to your Docker container and all future tasks on the cluster will also execute in the container.
```
Comment (concretevitamin) [inline docs/source/examples/docker-containers.rst]: A bit confused by this sentence. There's 1 container per node of the cluster, right? So "all future tasks on the cluster will also execute in the container." seems inaccurate in the multinode case (there's no single container but N containers). How to best rephrase?

**Review by concretevitamin — COMMENTED** [review]
Nice @cblmemo. Pushed some updates & left some comments.

File: docs/source/examples/docker-containers.rst (hunk 4 above)
Comment (cblmemo) [reply docs/source/examples/docker-containers.rst]: for private registries with containerized apps, the user still needs to manually login first. Let me add an example 🫡

File: docs/source/examples/docker-containers.rst (hunk 5 above)
Comment (cblmemo) [reply docs/source/examples/docker-containers.rst]: Now changed to:
```text
Any GPUs assigned to the task will be automatically mapped to your Docker container, and all subsequent tasks within the cluster will also run inside the container. In a multi-node scenario, the container will be launched on all nodes, and the corresponding node's container will be assigned for task execution.
```

**Comment by concretevitamin** [comment]
Pushed some updates. How does it look?

**Comment by cblmemo** [comment]
> Pushed some updates. How does it look?

It looks great! Let me add something like [this page](https://skypilot.readthedocs.io/en/latest/examples/syncing-code-artifacts.html), then it should be good to go 🫡

**Comment by cblmemo** [comment]
@concretevitamin Added a reference link and requirements to the docker image as runtime env. PTAL 🫡

**Review by concretevitamin — APPROVED** [review]
LGTM @cblmemo

**Pull request merged**